Resources improvements for native macOS app

Thomas-Vos opened the pull request:
See old PR for details: https://github.com/JetBrains/compose-multiplatform/pull/5169

~~Requires skiko update (https://github.com/JetBrains/compose-multiplatform-core/pull/1804)~~

Thomas-Vos commented:
@terrakok Skiko is now already updated on jb-main branch. Would it be possible to run CI on this PR to check if all is working?

Thomas-Vos commented:
@terrakok tested on an Intel Mac and the tests now pass with the updated skiko version:

![Screenshot 2025-02-10 at 00 33 52](https://github.com/user-attachments/assets/ed0a2acd-09df-489f-a41f-e31db93c20c0)


In gradle-plugins module I set `compose.version=1.8.0+dev2049` in `gradle.properties` and used local build of resources lib (set 0.1.0-SNAPSHOT version for resources in `ComposePlugin.kt`).

Thomas-Vos commented:
Could this be reviewed, please? Is there anything that needs to be changed?

terrakok commented:
`org.jetbrains.compose.test.tests.integration.ResourcesTest.testMultiModuleResources(kotlin=2.1.0, gradle=8.8, agp=8.5.0)` fails on Linux with:

```
Caused by: org.gradle.internal.resolve.ModuleVersionNotFoundException: Could not find me.sample.library:cmplib-macosarm64:1.0.
Searched in the following locations:
  - https://repo.maven.apache.org/maven2/me/sample/library/cmplib-macosarm64/1.0/cmplib-macosarm64-1.0.pom
  - https://dl.google.com/dl/android/maven2/me/sample/library/cmplib-macosarm64/1.0/cmplib-macosarm64-1.0.pom
  - https://packages.jetbrains.team/files/p/mpp/kgp-for-compose2/me/sample/library/cmplib-macosarm64/1.0/cmplib-macosarm64-1.0.pom
  - https://maven.pkg.jetbrains.space/public/p/compose/dev/me/sample/library/cmplib-macosarm64/1.0/cmplib-macosarm64-1.0.pom
  - https://maven.pkg.jetbrains.space/kotlin/p/kotlin/dev/me/sample/library/cmplib-macosarm64/1.0/cmplib-macosarm64-1.0.pom
  - file:/root/.m2/repository/me/sample/library/cmplib-macosarm64/1.0/cmplib-macosarm64-1.0.pom
  - file:/mnt/agent/temp/buildTmp/junit3414287715548457763/my-mvn/me/sample/library/cmplib-macosarm64/1.0/cmplib-macosarm64-1.0.pom
Required by:
    project :appModule > me.sample.library:cmplib:1.0
```

Thomas-Vos commented:
@igordmn @terrakok thanks for letting me know, opened a new PR: https://github.com/JetBrains/compose-multiplatform/pull/5245

It would be nice if TeamCity works on PRs as well, just like how Ktor does it. That would make contributing easier.

ubuntu/windows:

Reproduced on Ubuntu, caused by a bug in `checkExperimentalTargets.kt`, fixed by second commit in new PR.

macos:

My bad, I thought GitHub CI would override `compose.version` in gradle.properties just like TeamCity, but it does not. GitHub CI was broken in this PR so did not notice the issue, but seems to work again in PR and shows this is now fixed.

Now there is only failed test on GitHub CI due to MissingResourceException, which is because CI does not use latest version of resources component library, it has changes which are required for the test to succeed.
